### PR TITLE
Compact string format

### PR DESCRIPTION
**Pull request by jl-wynen**
Like #2556 but now merging into main

### CODE REVIEW

File: src/scipp/format/formatter.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    unt = "" if data.unit == Unit('dimensionless') else f" {data.unit}"
+
+    if compact:
+        # Iterate over array values to handle no- and infinite-precision cases
+        if var is None:
+            formatted = [_format(v) for v in val]
+        else:
+            formatted = [_format(*_round(v, e)) for v, e in zip(val, var)]
+        return f"{', '.join(formatted)}{unt}"
+        # return f"{_format(val, var)}{unt}"
```
Comment (nvaytet) [inline src/scipp/format/formatter.py:34]: Commented code

**Review by nvaytet — APPROVED** [review]

**Review by nvaytet — CHANGES_REQUESTED** [review]

File: tests/format/format_test.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    from numpy import array
+    array_variables = [(array([100, 20, 3]), None, 's', '100, 20, 3 s'),
+                       (array([100., 20.]), array([1., 2.]), 'm', '100.0(10), 20(2) m'),
+                       (array([9000., 800., 70., 6.]), array([100., 20., 3., 0.4]), '1',
+                        '9000(100), 800(20), 70(3), 6.0(4)'),
+                       (array([1., 2.,
+                               3.]), array([0., 1.,
+                                            0.2]), 'C', '1.0, 2.0(10), 3.0(2) C')]
+    for value, error, unit, expected in array_variables:
+        assert array_string(value, error, unit) == expected
```
[the remark this answers is not in the document]
Comment (jl-wynen) [reply tests/format/format_test.py:54]: Agree, and #2626. But I am not sure where to put the docs. A dedicated page seems overkill. At least until we have more options. Ideas?

[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply tests/format/format_test.py:54]: Maybe here: https://scipp.github.io/visualization/representations-and-tables.html? Possibly rename the page?

**Comment by jl-wynen** [comment]
Added documentation for it

**Comment by nvaytet** [comment]
The diff looks all messsed up.
Some release notes have disappeared and the notebooks contains some output?
```
"metadata": {
    "pycharm": {
     "name": "#%%\n",
     "is_executing": true
    }
   },
```
?

**Comment by jl-wynen** [comment]
Looks like I screwed up the merge. I force pushed to overwrite the bad merge commit.

And the notebook metadata should have been stripped by `nbstripout`. When running locally, it seems that it doesn't recurse into the right directory. Need to investigate...

**Comment by jl-wynen** [comment]
@nvaytet Any more comments?

**Review by nvaytet — APPROVED** [review]
Sorry i had not seen that you fixed the issues.

**Pull request merged**